### ISSUE TITLE
DATASET NELL

### ISSUE THREAD

**Issue by Jiang-X-Pro** [wangbo9719/StAR_KGC#12]
it went wrong when reproducing the results of NELL, usepacka was not found

**Comment by Jiang-X-Pro** [comment]
I ran StAR on NELL three times and could not reach the results shown in the paper 

10/11 11:31:06 AM: Hits right @1: 0.13901760889712697
10/11 11:31:06 AM: Hits right @5: 0.2882298424467099
10/11 11:31:06 AM: Hits right @10: 0.40593141797961074
10/11 11:31:06 AM: Mean rank right: 210.76181649675627
10/11 11:31:06 AM: Mean reciprocal rank right: 0.2222922814607432

10/11 07:36:31 PM: Hits right @1: 0.13855421686746988
10/11 07:36:31 PM: Hits right @5: 0.28776645041705284
10/11 07:36:31 PM: Hits right @10: 0.4040778498609824
10/11 07:36:31 PM: Mean rank right: 226.05421686746988
10/11 07:36:31 PM: Mean reciprocal rank right: 0.22114345866141052

10/12 09:55:17 AM: Hits right @1: 0.13994439295644115
10/12 09:55:17 AM: Hits right @5: 0.2882298424467099
10/12 09:55:17 AM: Hits right @10: 0.4082483781278962
10/12 09:55:17 AM: Mean rank right: 223.91890639481002
10/12 09:55:17 AM: Mean reciprocal rank right: 0.2234637227031943

**Comment by wangbo9719** [comment]
For the first problem, I have fixed it. Sorry for the mistake and thanks for your reminder. 

For the second question, I'll run the code and tell you the result. As a kind inquiry, did you follow all steps to reproduce the results?

**Comment by wangbo9719** [comment]
Below are my running results without any tuning:

Hits right @1: 0.16126042632066728
Hits right @5: 0.3438368860055607
Hits right @10: 0.44717330861909177
Mean rank right: 253.98285449490268
Mean reciprocal rank right: 0.2550246084722146

**Comment by Jiang-X-Pro** [comment]
I get the reformatted data by : python reformat_nell_one.py --data_dir path_to_downloaded --output_dir ./data/NELL_standard

and then use the following command to train and test:
 CUDA_VISIBLE_DEVICES=0 \
python run_link_prediction.py \
    --model_class bert \
    --do_train --do_eval --do_prediction \
    --warmup_proportion 0.1 \
    --learning_rate 5e-5 \
    --num_train_epochs 8. \
    --dataset NELL_standard \
    --max_seq_length 32 \
    --gradient_accumulation_steps 1 \
    --train_batch_size 16 \
    --eval_batch_size 128 \
    --logging_steps 100 \
    --eval_steps -1 \
    --save_steps 2000 \
    --model_name_or_path bert-base-uncased \
    --do_lower_case \
    --output_dir ./result/NELL_model \
    --num_worker 12 \
    --seed 42 \
    --fp16 \
    --cls_method cls \
    --distance_metric euclidean 

torch version is 1.6 and TITAN  RTX is used 

Are there any problems in the reproduction process?

**Comment by wangbo9719** [comment]
Yesterday, to reproduce the NELL-ONE result, I downloaded this repo on a new server and ran the above two commands as you did. The obtained results are similar to those reported in paper. 
My torch version is 1.8 and NVIDIA RTX 6000 is used. I do not think that it is the reason. 

A possible reason for the performance drop is the missing of the ''--fp16'' parameter. But you used it in your running. 

I have no idea about the drop. Maybe you can try more seeds. Seeds can affect the final result in many cases.

**Comment by Jiang-X-Pro** [comment]
I use a command for reproducing:
mv data/NELL/train_tasks.json.filepart  data/NELL/train_tasks.json
does it have any problem?

**Comment by wangbo9719** [comment]
It does not matter. I did it also.
